client/consensus/grandpa/sharedAuthoritySet.go

```diff
@@ -0,0 +1,48 @@
+// Copyright 2023 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package grandpa
+
+import (
+	"sync"
+)
+
+// SharedData Generic shared data structure
+type SharedData[T any] struct {
```

Comment:
I don't think we need to create another struct to safely read/write data, we can keep it simple by having the mutex as a property in the `SharedAuthoritySet`.

``` go
// SharedAuthoritySet A shared authority set
type SharedAuthoritySet struct {
        inner AuthoritySet
        mtx sync.Mutex <- mutex as a property
	// inner SharedData[AuthoritySet]
}
```

Reply:
Oh yeah we do not need it for safety, since substrate uses the SharedData type in other parts of the code I initially thought that it would be good to replicate, but given how much less overhead we have for this I think you are right on this.

Will ping @timwu20 just to make sure this approach is okay before removing the type

Reply:
Yea I don't think we need this.  We can just use a mutex in the `AuthoritySet` itself and make it thread safe.  Or we don't need the generic type of `SharedData`.

Reply:
I updated the code to remove the SharedData Ttype, and replaced it with what @EclesioMeloJunior recommended.